sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogV2Util.scala

```diff
@@ -598,10 +598,28 @@ private[sql] object CatalogV2Util {
       //       data unchanged and let the data reader to return "exist default" for missing
       //       columns.
       val existingDefault = Literal(default.getValue.value(), default.getValue.dataType()).sql
-      f.withExistenceDefaultValue(existingDefault).withCurrentDefaultValue(default.getSql)
+      f.withExistenceDefaultValue(existingDefault).withCurrentDefaultValue(toSql(defaultValue))
     }.getOrElse(f)
   }
 
+  private def toSql(defaultValue: DefaultValue): String = {
```

Comment:
style question: did we consider putting this logic in DefaultValue?

Reply:
`DefaultValue` is a user-facing DS v2 API. We should put these internal util functions in internal classes.